### PR TITLE
Slider animations

### PR DESCRIPTION
**Pull request by moduscreate**
Introduced slider animations.

```
'use strict';

var React = require('react');
var ReactNative = require('react-native');
var Slider = require('react-native-slider');
var {
  AppRegistry,
  StyleSheet,
  Text,
  ScrollView,
  View,
  SliderIOS,
  Easing
} = ReactNative;

var SliderExample = React.createClass({
  getInitialState() {
    return {
      value1: -3,
      value2: -10
    };
  },
  componentDidMount() {
      setTimeout(() => {  
        this.setState({ value : 5})
      }, 1000);  
  },

  render() {
    var state = this.state,
        value1 = state.value1,
        value2 = state.value2;

    return (
      <View style={styles.container}>

          <View>
            <View style={styles.titleContainer}>
              <Text style={styles.caption} numberOfLines={1}>Drag this slider</Text>
              <Text style={styles.value} numberOfLines={1}>{value1}</Text>
            </View>
            <Slider
              minimumValue={-10}
              maximumValue={10}
              step={1}
              value={value1}
              trackStyle={customStyles5.track}
              thumbStyle={customStyles5.thumb}
              onValueChange={(val) => this.setState({value1:val, value2:val})}
            />
          </View>
          
          <View>
            <View style={styles.titleContainer}>
              <Text style={styles.caption} numberOfLines={1}>Slider should animate</Text>
              <Text style={styles.value} numberOfLines={1}>{value2}</Text>
            </View>
            <Slider
              minimumValue={-10}
              maximumValue={10}
              step={1}
              value={value2}
              trackStyle={customStyles5.track}
              thumbStyle={customStyles5.thumb}
              onValueChange={(val) => this.setState({value1:val, value2,val})}
              animateTransitions={true}
              animationType={'spring'}
            />
          </View>
        
      </View>
    );
  },
});

var styles = StyleSheet.create({
  container: {
    margin: 20,
    paddingVertical : 150,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
  },
  titleContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  caption: {
    //flex: 1,
  },
  value: {
    flex: 1,
    textAlign: 'right',
    marginLeft: 10,
  }
});


var customStyles5 = StyleSheet.create({
  track: {
    height: 18,
    borderRadius: 1,
    backgroundColor: '#d5d8e8',
  },
  thumb: {
    width: 20,
    height: 30,
    borderRadius: 1,
    backgroundColor: '#838486',
  }
});

AppRegistry.registerComponent('rnslider', () => SliderExample);

```

### CODE REVIEW

File: src/Slider.js [hunk 1]
```diff
@@ -151,6 +183,7 @@ var Slider = React.createClass({
       thumbTintColor: '#343434',
       thumbTouchSize: {width: 40, height: 40},
       debugTouchArea: false,
+      animationType: 'spring'
```
Comment (jeanregisser) [inline src/Slider.js]: Could you set the default to `timing` instead?
I feel that's what most people would expect.

File: src/Slider.js [hunk 2]
```diff
@@ -165,10 +198,17 @@ var Slider = React.createClass({
     });
   },
   componentWillReceiveProps: function(nextProps) {
-    var oldValue = this.props.value;
-    var newValue = nextProps.value;
+    var props    = this.props,
+        oldValue = props.value,
+        newValue = nextProps.value;
+
```
Comment (jeanregisser) [inline src/Slider.js]: nit: I prefer not grouping individual declarations (use destructuring if needed).

File: src/Slider.js [hunk 3]
```diff
@@ -9,14 +9,15 @@ import {
...
 
-var shallowCompare = require('react-addons-shallow-compare');
-var styleEqual = require('style-equal');
+const shallowCompare = require('react-addons-shallow-compare'),
+      styleEqual = require('style-equal');
 
-var TRACK_SIZE = 4;
-var THUMB_SIZE = 20;
+const TRACK_SIZE = 4,
+      THUMB_SIZE = 20;
```
Comment (jeanregisser) [inline src/Slider.js]: nit: no need to change those, I prefer the way it was previously. But we can keep const.

File: src/Slider.js [hunk 4]
```diff
@@ -364,6 +404,19 @@ var Slider = React.createClass({
     this.state.value.setValue(value);
   },
 
+  _setCurrentValueAnimated(value: number) {
+    var props           = this.props,
+        animationType   = props.animationType,
+        animationConfig = Object.assign(
```
Comment (jeanregisser) [inline src/Slider.js]: nit: same as above, use multiple declarations (and destructuring if needed).

**Comment by jeanregisser** [comment]
Excellent addition @jaygarcia! 👍

File: src/Slider.js (hunk 2 above)
Comment (jaygarcia) [inline src/Slider.js]: The pattern of development that I've used is geared towards better magnification & faster processing by the interpreter.

File: src/Slider.js (hunk 3 above)
Comment (jaygarcia) [inline src/Slider.js]: I'll revert.

File: src/Slider.js (hunk 1 above)
Comment (jaygarcia) [inline src/Slider.js]: Sure thing.

**Pull request merged**